include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -513,6 +513,71 @@ ORT_API_STATUS(OrtGetValueCount, const OrtValue* value, size_t* out);
 ORT_API_STATUS(OrtCreateValue, OrtValue** const in, int num_values, enum ONNXType value_type,
                OrtValue** out);
 
+/*
+ * EXPERIMENTAL APIS - Subject to change. Released as a preview to get feedback and enable early testing
+*/
+
+/*
+ * Steps to use a custom op:
+ *   1 Create an OrtCustomOpDomain with the domain name used by the custom ops
+ *   2 Create an OrtCustomOp structure for each op and add them to the domain
+ *   3 Call OrtAddCustomOpDomain to add the custom domain of ops to the session options
+*/
+struct OrtKernelInfo;
+typedef struct OrtKernelInfo OrtKernelInfo;
+
+/*
+ * These allow reading node attributes during kernel creation
+*/
+ORT_API_STATUS(OrtKernelInfoGetAttribute_float, _In_ OrtKernelInfo* info, _In_ const char* name, _Out_ float* out);
+ORT_API_STATUS(OrtKernelInfoGetAttribute_int64, _In_ OrtKernelInfo* info, _In_ const char* name, _Out_ int64_t* out);
+
+/*
+ * The OrtCustomOp structure defines a custom op's schema and it's kernel callbacks. The callbacks are filled in by
+ * the implementor of the custom op.
+*/
+struct OrtCustomOp {
+  uint32_t version;  // Initialize to ORT_API_VERSION
```

Comment:
dumb question please. what's the usage of this "version"? looks to me it's useless.

Reply:
If we ever want to update this structure in the C API we need to know how to recognize the old vs the new versions used by code that wasn't recompiled. This lets us know what version of the structure they're using and we can handle it internally.

Reply:
I'm seeing ORT_API_VERSION is the one used for this purpose - "we need to know how to recognize the old vs the new versions used by code that wasn't recompiled"

Do we need one in each structure please?

Reply:
It's only needed in structs defined inside the header file. So just OrtAllocator and now this.